### ISSUE TITLE
Question: Why filter `show_advanced_plugins` only when `is_admin`

### ISSUE THREAD

**Issue by tangrufus** [roots/bedrock-autoloader#18]
* [x] I've read the [guidelines for Contributing to Roots Projects](https://github.com/roots/.github/blob/master/CONTRIBUTING.md)
* [x] This request isn't a duplicate of an existing issue
* [x] I've read the [docs](https://roots.io/documentation/) and followed them (if applicable)
* [x] This is not a personal support request that should be posted on the [Roots Discourse](https://discourse.roots.io/) community

https://github.com/roots/bedrock-autoloader/blob/39dab98cf2fea1b1debb40c6d64ee6947ce98d64/src/Autoloader.php#L56-L58

The `is_admin` check was introduced since the [beginning](https://github.com/roots/bedrock/commit/82a9c4338cf7da670756ce7a8d4f4ccd22511d1b#diff-ff1471ea7fc3cd4901ed95b84216a0edR33-R35).

Question: Why the `is_admin` check?
In other words, in what case `show_advanced_plugins` is triggered in non-admin context?

**Comment by retlehs** [comment]
I guess the only benefit is not registering an admin-only filter on front-end/CLI requests?

**Comment by tangrufus** [comment]
It's been too long I forgot what problems I encountered.
But no other persons reported issues. I guess it is fine.
